charts/datadog/templates/cluster-agent-rbac.yaml

```
@@ -303,6 +233,23 @@ subjects:
     name: {{ template "datadog.fullname" . }}-cluster-agent
     namespace: {{ .Release.Namespace }}
 ---
+{{- if and .Values.datadog.kubeStateMetricsCore.enabled (not .Values.datadog.kubeStateMetricsCore.useClusterCheckWorkers) }}
+apiVersion: {{ template "rbac.apiVersion" . }}
+kind: ClusterRoleBinding
+metadata:
+  labels:
+{{ include "datadog.labels" . | indent 4 }}
+  name: {{ template "datadog.fullname" . }}-cluster-agent-ksm-core
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: {{ template "datadog.fullname" . }}-ksm-core
+subjects:
+  - kind: ServiceAccount
+    name: {{ template "datadog.fullname" . }}-cluster-agent
+    namespace: {{ .Release.Namespace }}
+---
+{{- end }}
```

Following the above comment, as `---` is the marker of the beginning of a yaml document, it looks more natural to have the `---` of the next document outside this `{{- if` block.

```suggestion
{{- end }}
---
```